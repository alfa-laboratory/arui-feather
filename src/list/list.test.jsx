/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow } from 'enzyme';

import List from './list';

describe('list', () => {
    it('should render without problem', () => {
        let list = shallow(<List />);

        expect(list).toMatchSnapshot();
        expect(list.hasClass('list')).toBe(true);
    });

    it('should render ul type without problem', () => {
        const ITEMS_UL = [
            {
                key: 'one',
                value: 'Apple'
            },
            {
                key: 'two',
                value: 'Orange'
            },
            {
                key: 'three',
                value: 'Banana'
            }
        ];

        let list = shallow(<List items={ ITEMS_UL } type='ordered' />);

        expect(list).toMatchSnapshot();
        expect(list.hasClass('list_type_ordered')).toBe(true);
    });
});
