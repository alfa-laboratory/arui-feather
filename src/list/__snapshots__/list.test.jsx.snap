// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should render ul type without problem 1`] = `
<ol
  className="list list_type_ordered list_theme_alfa-on-white"
>
  <li
    className="list__item"
    key="item-one"
  >
    Apple
  </li>
  <li
    className="list__item"
    key="item-two"
  >
    Orange
  </li>
  <li
    className="list__item"
    key="item-three"
  >
    Banana
  </li>
</ol>
`;

exports[`list should render without problem 1`] = `
<ul
  className="list list_theme_alfa-on-white"
/>
`;
